### ISSUE TITLE
Migrate `TABLE` related DDL statements to new planner

### ISSUE THREAD

**Issue by leiysky** [datafuselabs/databend#5774]
Statements:

- [x] `CREATE TABLE` #5771 
- [x] `DROP TABLE`
- [x] `ALTER TABLE`
- [x] `UNDROP TABLE`
- [x] `RENAME TABLE`
- [x] `TRUNCATE TABLE`
- [x] `OPTIMIZE TABLE`
- [ ] `DESCRIBE TABLE`
- [ ] `SHOW CREATE DATABASE`
- [ ] `SHOW CREATE TABLE`
- [ ] `SHOW TABLES`
- [ ] `SHOW DATABASES`
- [ ] `SHOW TABLE STATUS`

**Comment by andylokandy** [comment]
Blocked by https://github.com/datafuselabs/databend/issues/5778, because `ALTER TABLE CLUSTER BY {expr}` has to analyze the expr using `ExpressionAnalyzer`, which is under migration in #5778.

**Comment by AngleNet** [comment]
It's fine. I will pick another issue.
